### PR TITLE
[release-4.11] OCPBUGS-3507: On-prem: Ensure resolv-prepender respects NM dispatcher timeout

### PR DESCRIPTION
**Pull request by martinkennelly**
Not a clean cherry-pick of 1b97dcc2390e3f76f39dfc732c9124030c5f7c5e from 4.12

Needed to replace 2 lines with:
```
                "{{ onPremPlatformAPIServerInternalIP . }}" \
                "{{ onPremPlatformIngressIP . }}")
```

### CODE REVIEW

**Comment by martinkennelly** [comment]
/retest

# CI Failures
## e2e-metal-ipi
Infra setup failure unrelated to this PR

## e2e-metal-ipi-ovn-dualstack
Opened an issue in CMO to tackle this: https://issues.redhat.com/browse/OCPBUGS-3923
Unrelated to this PR.

**Comment by martinkennelly** [comment]
@vrutkovs your command failed. How can we test this against okd?

File: templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml
Comment (vrutkovs) [inline templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml:107]:
```suggestion
                    if [[ ! -f "${KNICONFPATH} ]] || [[ ! -f "${KNICONFDONEPATH} ]] || ! cmp --silent "${KNICONFPATH}" "${KNICONFTMPPATH}"; then
```

**Review by vrutkovs — CHANGES_REQUESTED** [review]
/hold

This will break OKD

**Comment by vrutkovs** [comment]
>How can we test this against okd?

Weird, release-4.11 is the only branch which is not covered. Adding OKD tests in https://github.com/openshift/release/pull/34283

File: templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml
Comment (vrutkovs) [inline templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml:110]:
```suggestion
                        mv -f "${KNICONFTMPPATH}" "${KNICONFPATH}"
                        chown systemd-resolve:systemd-resolve "${KNICONFPATH}"
                        chcon system_u:object_r:etc_t:s0 "${KNICONFPATH}"
```
tmp file has invalid permissions and SELinux label

**Comment by yuqi-zhang** [comment]
The 4.12 branch commit referenced (https://github.com/openshift/machine-config-operator/commit/1b97dcc2390e3f76f39dfc732c9124030c5f7c5e) doesn't seem to have some of the fixes above (e.g. selinux). Should we fix that in 4.12 first before continuing with this backport?

**Comment by vrutkovs** [comment]
Yeah, still fixing master/release-4.12

File: templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml
Comment (vrutkovs) [reply templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml:110]: Or simply `cat "${KNICONFTMPPATH}" > "${KNICONFPATH}"`

**Comment by vrutkovs** [comment]
/hold cancel

OKD vsphere install passes (fails on imagestream import which is a known issue)

**Comment by martinkennelly** [comment]
@vrutkovs I cherry-picked your commits from 4.12. Conflict observed for commit 0e80c65d7c8bac64e5214af4ba4aae67043c857f was for these two lines:
```
      "{{ onPremPlatformAPIServerInternalIP . }}" \
      "{{ onPremPlatformIngressIP . }}")
```

No conflict for c5a078fbb36c753efdd01b3f20b67a226d52cd88

**Comment by martinkennelly** [comment]
/retest
Unrelated test failures.

**Comment by martinkennelly** [comment]
@vrutkovs can i get your review (and hopeful blessing) for this?

**Review by yuqi-zhang — APPROVED** [review]
/label backport-risk-assessed

/hold for QE approval

Would this be verifiable by someone (Itay maybe?) pre-merge?

**Comment by martinkennelly** [comment]
> /label backport-risk-assessed
> 
> /hold for QE approval
> 
> Would this be verifiable by someone (Itay maybe?) pre-merge?

Ill ask him can he do it on the jira issue. Thank you.

**Comment by martinkennelly** [comment]
@anuragthehatter PTAL
Testing will be done post merge using a nightly after consulting with QE due to the complex setup required to validate this fix. See the associated issue comments for more details.

**Comment by martinkennelly** [comment]
@anuragthehatter PTAL. Thank you.

**Pull request merged**
